enhancements/network/communication-ingress-flows-matrix.md

```diff
@@ -0,0 +1,198 @@
+single node deployments) and Operators.
+
+## Motivation
+
+Security-conscious customers need OpenShift flows matrix for regulatory reasons 
+and/or to implement firewall rules to restrict traffic to the minimum set of
+required flows only, on-node firewall or external.
+
+### User Stories
+
+- As an OpenShift cluster administrator, I want documentation on the expected 
+  flows of traffic incoming to every OpenShift installation so I can set up 
+  firewall rules such as nftables, NGFW, etc. to restrict traffic to the 
+  minimum required set of flows only.
+
+- As an OpenShift cluster administrator, I want a "one-click" feature to 
+  generate an accurate, relevant, matrix for the running OpenShift cluster.
+
+- As a testing engineer, I want a machine-readable ingress traffic matrix 
+  so that I can analyze and compare with tools such as nmap.
+
+### Goals
+
+- Provide a mechanism to automatically generate an accurate and up-to-date 
+  OpenShift communication ingress flows matrix.
+
+- Keep the communication matrix documented in OpenShift release documents
+  updated and validate it does not change.
+
+### Non-Goals
+
+- Egress traffic.
+
+## Proposal
+
+We introduce a new module to generate an accurate and up-to-date OpenShift 
+communication ingress flows matrix.
+
+- A communication matrix describing the expected flows of incoming traffic will 
+  be included in every OpenShift release documentation.
+
+- A new `oc` command will be added to generate a current snapshot of known 
+  listening ports in a running cluster, `oc adm generate-commatrix`. #Sabina: what should be the command name?
+
+- A new option will be added to OpenShift web console to generate an up-to-date
+  communication matrix.
+
+### Workflow Description
+
+An OpenShift administrator would like to get an accurate and up-to-date OpenShift 
+communication ingress flows matrix.
+
+- The admin reviews OpenShift release documentation to get the included communication
+  matrix describing the expected flows of incoming traffic.
+
+- The admin uses the OpenShift command-line interface (CLI) to generate an up-to-date 
+  communication matrix using the following command `oc adm generate-commatrix`.
+
+- Optionally, the admin uses the OpenShift web console to generate an up-to-date
+  communication matrix.
+
+### API Extensions
+N/A
+
+### Implementation Details/Notes/Constraints
+
+A new helper module `node-comm-lib` will be added to `openshift/library-go`. #Sabina: what should be the module name?
+The module will be used to create a communication ingress flows matrix 
+from the existing `EndpointSlices` in the OpenShift cluster.
+
+`EndpointSlices` group network endpoints together by unique combinations of 
+protocol, port number, and `Service` name and therefore can be used as the 
+source of truth for cluster traffic. For that to work, every application that 
+attracts ingress traffic running in the cluster should have a `Service` object,
```

Comment:
they can also create endpointslices directly if they are doing "weird" stuff (like privileged parts of the platform that do not create a service, but are still listening on host net ports)
I'm basing this on wording found here
https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/#management
that explain it's ok to create those